File: app/src/main/java/io/wax911/emojifysample/MainActivity.java
package io.wax911.emojifysample;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.AppCompatEditText;
import android.text.Editable;
import android.text.Spanned;
import android.text.SpannedString;
import android.view.View;
import android.widget.Toast;

import io.wax911.emojify.EmojiUtils;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    private AppCompatEditText editText;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        editText = findViewById(R.id.edit_text);

        findViewById(R.id.toEmoji).setOnClickListener(this);
        findViewById(R.id.toHtml).setOnClickListener(this);
        findViewById(R.id.toHexHtml).setOnClickListener(this);
        findViewById(R.id.toShortCode).setOnClickListener(this);
    }

    @Override
    public void onClick(View view) {
        Editable textContent;
        if((textContent = editText.getEditableText()) == null) {
            Toast.makeText(this, "You must first enter some text", Toast.LENGTH_SHORT).show();
            return;
        }
        Spanned convertedText = null;
        switch (view.getId()) {
            case R.id.toEmoji:
                // alternatively you could convert your hexHtml to emoji by using Html.fromHtml()
                //e.g. convertedText = Html.fromHtml(EmojiUtils.hexHtmlify(textContent.toString()));
                convertedText = new SpannedString(EmojiUtils.emojify(textContent.toString()));
                break;
            case R.id.toHtml:
                // alternatively you could convert your hexHtml to emoji by using Html.fromHtml()
                //e.g. convertedText = Html.fromHtml(EmojiUtils.htmlify(textContent.toString()));
                convertedText = new SpannedString(EmojiUtils.htmlify(textContent.toString()));
                break;
            case R.id.toHexHtml:
                convertedText = new SpannedString(EmojiUtils.hexHtmlify(textContent.toString()));
                break;
            case R.id.toShortCode:
                convertedText = new SpannedString(EmojiUtils.shortCodify(textContent.toString()));
                break;
        }
        editText.setText(convertedText);
    }
}

File: app/src/main/java/io/wax911/emojifysample/MainActivity.kt
package io.wax911.emojifysample

import android.os.Bundle
import android.text.Editable
import android.text.Spanned
import android.text.SpannedString
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import io.wax911.emojify.parser.parseToAliases
import io.wax911.emojify.parser.parseToHtmlDecimal
import io.wax911.emojify.parser.parseToHtmlHexadecimal
import io.wax911.emojify.parser.parseToUnicode
import io.wax911.emojifysample.databinding.ActivityMainBinding
import io.wax911.emojifysample.ext.emojiManager

class MainActivity : AppCompatActivity(), View.OnClickListener {

    private val binding: ActivityMainBinding by lazy {
        ActivityMainBinding.inflate(layoutInflater)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(binding.root)
    }

    override fun onPostCreate(savedInstanceState: Bundle?) {
        super.onPostCreate(savedInstanceState)
        binding.containMain.toEmoji.setOnClickListener(this)
        binding.containMain.toHtml.setOnClickListener(this)
        binding.containMain.toHexHtml.setOnClickListener(this)
        binding.containMain.toShortCode.setOnClickListener(this)
    }

    override fun onClick(view: View) {
        val textContent: Editable? = binding.containMain.editText.editableText
        if (textContent == null) {
            Toast.makeText(this, "You must first enter some text", Toast.LENGTH_SHORT).show()
            return
        }
        var convertedText: Spanned? = null
        when (view.id) {
            R.id.toEmoji ->
                // alternatively you could convert your hexHtml to emoji by using Html.fromHtml()
                //e.g. convertedText = Html.fromHtml(EmojiParser.parseToHtmlHexadecimal(textContent.toString()));
                convertedText = SpannedString(emojiManager().parseToUnicode(textContent.toString()))

            R.id.toHtml ->
                // alternatively you could convert your hexHtml to emoji by using Html.fromHtml()
                //e.g. convertedText = Html.fromHtml(EmojiParser.parseToHtmlDecimal(textContent.toString()));
                convertedText =
                    SpannedString(emojiManager().parseToHtmlDecimal(textContent.toString()))

            R.id.toHexHtml ->
                convertedText =
                    SpannedString(emojiManager().parseToHtmlHexadecimal(textContent.toString()))

            R.id.toShortCode ->
                convertedText = SpannedString(emojiManager().parseToAliases(textContent.toString()))
        }
        binding.containMain.editText.setText(convertedText)
    }

    override fun onDestroy() {
        binding.containMain.toEmoji.setOnClickListener(null)
        binding.containMain.toHtml.setOnClickListener(null)
        binding.containMain.toHexHtml.setOnClickListener(null)
        binding.containMain.toShortCode.setOnClickListener(null)
        super.onDestroy()
    }
}
